### ISSUE TITLE
Results require runAfter when embedding Pipeline spec

### ISSUE THREAD

**Issue by bobcatfish** [tektoncd/pipeline#2463]
# Expected Behavior

Embedding a Pipeline spec into a PipelineRun should not change the behaviour of results.

# Actual Behavior

If you remove the `runAfter` clause from [this example](https://github.com/tektoncd/pipeline/blob/master/examples/v1beta1/pipelineruns/pipelinerun-results-with-params.yml), you get an error caused by the controller mistakenly starting both Tasks simultaneously.

```yaml
apiVersion: tekton.dev/v1beta1
kind: PipelineRun
metadata:
  generateName: test-case
  name: test-case-run
spec:
  params:
  - name: prefix
    value: prefix
  pipelineSpec:
    params:
    - name: prefix
    tasks:
    - name: generate-suffix
      taskSpec:
        results:
        - name: suffix
        steps:
        - name: generate-suffix
          image: alpine
          script: |
            echo -n "suffix" > $(results.suffix.path)
    - name: do-something
      taskSpec:
        params:
        - name: arg
        steps:
        - name: do-something
          image: alpine
          script: |
            echo "$(params.arg)"
      params:
      - name: arg
       # this should be enough to imply to the controller that this Task should run
       # after generate-suffix, but with runAfter removed, the controller starts it at
       # the same time as generate-suffix
        value: "$(params.prefix):$(tasks.generate-suffix.results.suffix)"
```

# Steps to Reproduce the Problem

1. Apply the above yaml and look at the pipelinerun

# Additional Info

Interestingly this doesnt happen when not embedding the pipelinespec, e.g. this other example does not have the same problem https://github.com/tektoncd/pipeline/blob/master/examples/v1beta1/pipelineruns/pipelinerun-results.yaml

- Tekton Pipeline version: HEAD @ 1346656111decda59d4c36280b2cc828127a37e1

**Comment by bobcatfish** [comment]
Working on a fix for this now

/assign bobcatfish

**Comment by bobcatfish** [comment]
Fixed in #2471
